differentiation.go

```diff
@@ -164,22 +163,19 @@ func Backpropagate(outputs, gradOutputs, wrt Nodes) (retVal Nodes, err error) {
 
 	var sortedNodes Nodes
 	if sortedNodes, err = Sort(g); err != nil {
-		err = errors.Wrap(err, sortFail)
-		return
+		return nil, errors.Wrap(err, sortFail)
```

Comment:
I think this may be better:

```go
return nil, errors.Wrapf(err, sortFail, "Backprop")
```

This way, the user doesn't have to do the whole `Cause` thing to quickly figure out where the error happened. Thoughts?